plugins/InspectionGadgets/src/com/intellij/codeInspection/RedundantCompareToJavaTimeInspection.java

```diff
@@ -0,0 +1,139 @@
+import com.intellij.openapi.util.NlsSafe;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.*;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.psi.util.PsiUtil;
+import com.intellij.util.ObjectUtils;
+import com.siyeh.InspectionGadgetsBundle;
+import com.siyeh.ig.callMatcher.CallMatcher;
+import com.siyeh.ig.psiutils.CommentTracker;
+import com.siyeh.ig.psiutils.ExpressionUtils;
+import org.jetbrains.annotations.NotNull;
+
+public class RedundantCompareToJavaTimeInspection extends AbstractBaseJavaLocalInspectionTool implements CleanupLocalInspectionTool {
+  private static final CallMatcher COMPARE_TO_METHODS = CallMatcher.anyOf(
+    CallMatcher.instanceCall(CommonClassNames.JAVA_TIME_LOCAL_DATE, "compareTo").parameterTypes("java.time.chrono.ChronoLocalDate"),
+    CallMatcher.instanceCall(CommonClassNames.JAVA_TIME_LOCAL_TIME, "compareTo").parameterTypes(CommonClassNames.JAVA_TIME_LOCAL_TIME),
+    CallMatcher.instanceCall(CommonClassNames.JAVA_TIME_LOCAL_DATE_TIME, "compareTo")
+      .parameterTypes("java.time.chrono.ChronoLocalDateTime"),
+    CallMatcher.instanceCall(CommonClassNames.JAVA_TIME_OFFSET_TIME, "compareTo").parameterTypes(CommonClassNames.JAVA_TIME_OFFSET_TIME),
+    CallMatcher.instanceCall(CommonClassNames.JAVA_TIME_OFFSET_DATE_TIME, "compareTo")
+      .parameterTypes(CommonClassNames.JAVA_TIME_OFFSET_DATE_TIME)
+  );
+
+  @Override
+  public @NotNull String getID() {
+    return "RedundantCompareToJavaTimeStatement";
+  }
+
+  @NotNull
+  @Override
+  public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+    return new JavaElementVisitor() {
+      @Override
+      public void visitMethodCallExpression(@NotNull PsiMethodCallExpression call) {
+        if (!COMPARE_TO_METHODS.test(call)) return;
+        final PsiExpression qualifierExpression = call.getMethodExpression().getQualifierExpression();
+        if (qualifierExpression == null) {
+          return;
+        }
+        final PsiType argumentType = call.getArgumentList().getExpressionTypes()[0];
+        if (!argumentType.equals(qualifierExpression.getType())) {
+          return;
+        }
+
+        PsiBinaryExpression binOp = ObjectUtils.tryCast(PsiUtil.skipParenthesizedExprUp(call.getParent()), PsiBinaryExpression.class);
+        if (binOp == null) return;
+        RelationType relationType = DfaPsiUtil.getRelationByToken(binOp.getOperationTokenType());
+        if (relationType == RelationType.IS || relationType == RelationType.IS_NOT) {
+          return;
+        }
+        if (relationType == null) return;
+        if (ExpressionUtils.isZero(binOp.getLOperand())) {
+          relationType = relationType.getFlipped();
+          if (relationType == null) return;
+        }
+        else if (!ExpressionUtils.isZero(binOp.getROperand())) {
+          return;
+        }
+        holder.registerProblem(call, InspectionGadgetsBundle.message("redundant.call.problem.descriptor"),
```

Comment:
There is an easier way to highlight only the method identifier, without specifying a separate `TextRange`. See for example com.intellij.codeInspection.java18api.OptionalGetWithoutIsPresentInspection.

Reply:
Changed. Thank you!